intake_cesm/core.py

```diff
@@ -141,16 +141,44 @@ def results(self):
     def _open_dataset(self):
         url = self.urlpath
         kwargs = self._kwargs
+
+        query = dict(self.query)
         if "*" in url or isinstance(url, list):
-            _open_dataset = xr.open_mfdataset
             if "concat_dim" not in kwargs.keys():
                 kwargs.update(concat_dim=self.concat_dim)
             if self.pattern:
                 kwargs.update(preprocess=self._add_path_to_ds)
-        else:
-            _open_dataset = xr.open_dataset
 
-        self._ds = _open_dataset(url, chunks=self.chunks, **kwargs)
+            ensembles = self.query_results.ensemble.unique()
+            variables = self.query_results.variable.unique()
+
+            ds_ens_list = []
+            for ens_i in ensembles:
+                query['ensemble'] = ens_i
+
+                dsi = xr.Dataset()
+                for var_i in variables:
+
+                    query['variable'] = var_i
+                    urlpath_ei_vi = get_subset(self.collection, query).files.tolist()
+                    dsi = xr.merge(
+                        (
+                            dsi,
+                            xr.open_mfdataset(
+                                urlpath_ei_vi,
+                                data_vars=[var_i],
+                                chunks=self.chunks,
+                                **kwargs
+                            ),
+                        )
+                    )
+
+                    ds_ens_list.append(dsi)
+
+            self._ds = xr.concat(ds_ens_list, dim='ens',
+                                 data_vars=variables)
+        else:
+            self._ds = _open_dataset(url, chunks=self.chunks, **kwargs)
```

Comment:
@matt-long, I think this needs to be changed to `self._ds = xr.open_dataset(url, chunks=self.chunks, **kwargs)`

Reply:
done.  Reopening PR